Reloop Mixtour: initial implementation

absorb-it opened the pull request:
This is a implementation for the midi-controller Reloop Mixtour. It is a two deck controller and you configure the Decks for which to use this controller, if you have two of them you can control 4 decks with it.

Related documentation is at https://github.com/mixxxdj/manual/pull/825

Commit:
Reloop Mixtour: initial implementation

Commit:
Reloop Mixtour: added show maximized Library

added feature for Back button to toggle display of maximized Library

Commit:
Reloop Mixtour: cleanup of rotatySelector implementation

acolombier commented on res/controllers/Reloop-Mixtour-scripts.js:
```diff
@@ -0,0 +1,316 @@
+    if (this.connections[0] === undefined &&      // <-- added this condition
+        undefined !== this.group &&
+        undefined !== this.outKey &&
+        undefined !== this.output &&
```
Let's keep the comparison consistent and have the variable on the left 

```suggestion
    if (this.connections[0] === undefined &&      // <-- added this condition
        this.group !== undefined &&
        this.outKey !== undefined &&
        this.output !== undefined &&
```

absorb-it replied:
Hmm, get your point, but it's copy/pasted from the original implementation and by modifying more lines than required it makes the code harder to read/compare/understand. Therefore I prefer leaving this for now and probably adding a pull-request for midi-components-0.0.js before adaptation.

acolombier reviewed (changes requested):
Small nit regarding the use of the `midi-components`'s connections list. For context, this array stores active connection, so upon disconnect, you should remove it, instead of undefining it

Looks good otherwise, though haven't tested it!

absorb-it commented on res/controllers/Reloop-Mixtour-scripts.js:
```diff
@@ -0,0 +1,316 @@
+// original connect function will override connections[0] and
+// this way looses control of existing connection
+components.Component.prototype.connect = function() {
+    if (this.connections[0] === undefined &&      // <-- added this condition
```
The original code uses this comparison in line 131 https://github.com/mixxxdj/mixxx/blob/main/res/controllers/midi-components-0.0.js#L115C1-L137C1 - using the same style imho. makes sense.

Commit:
Reloop Mixtour: code cleanup

absorb-it committed:
Reloop Mixtour: bugfix Mixer selection

group indications should reflect the currentDeck, not the midi channel. Thanks for @andrasore for the  fix.

Co-authored-by: András Sőre <andras08@gmail.com>

Commit:
Reloop Mixtour: bugfix FX selection

group indications should reflect the currentDeck, not the midi channel.
Thanks for @andrasore for the pointing to the issue.

andrasore reviewed (changes requested):
Hi @absorb-it ! Sorry for the late reply. I've done a quick testing of the mapping fixes. I've noticed two things at a glance:

- When pressing the right "c" button on the controller (for putting the right side control buttons into cue mode), the second button (labeled "sync") does not create a cue. An error dialog is shown
- I think the FX buttons' main functionality should be switching between filter mode and selected quick FX for the channel. This is consistent with the buttons' labels, and is encouraged as per the contributing guidelines. Maybe shift + fx could select the next FX in the quick effects selector list.

Commit:
Reloop Mixtour: bugfix hotcue btn right deck

fixed boken button, thanks to andrasore for finding he bug!

Commit:
Reloop Mixtour: Changed Filter/FX Btn behaviour

Filter/FX button now changes between the Quickfilter number 1 (best set
this to a plain Filter) and other FX - as labeled on the device. Thanks
to @andrasore for code and testing.

andrasore commented:
Okay, sure! ☺️

> *absorb-it* left a comment (mixxxdj/mixxx#15790)
> <https://github.com/mixxxdj/mixxx/pull/15790#issuecomment-4057812779>
>
> Hi @andrasore <https://github.com/andrasore> ,
>
> I've tested the changes in the included js file. Everything is working
> correctly, and the code looks nicer too! LGTM
>
> committed with e08506d
> <https://github.com/mixxxdj/mixxx/commit/e08506d6626c25b37ffe43b7e35be7f9755c65dd>
> .
>
> I like to mention your work in the midi header author line, if this is ok
> for you. (as of now: Rene (mixxx at absorb.it), zfhrp). Can I add your
> name and email?
>
> Regards, Rene
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/15790#issuecomment-4057812779>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD2DZGE7FTRFBXEZ5S5VS3D4QRVPDAVCNFSM6AAAAACQCT5JQSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHM2DANJXHAYTENZXHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Commit:
Reloop Mixtour: added co-author

absorb-it commented:
Hi, I won't fix any hallucinated issues brought in by KI. Nor will I improve the quality of this KI with my comments on it's suggestions. If anybody else likes to take the time and go to this, feel free.

Rene
